src/js/react/media-store.tsx

```diff
@@ -0,0 +1,116 @@
+import { createContext, useContext, useEffect, useMemo } from 'react';
```

Comment:
hmm, I think for now we were using TS jsdocs right?

does this file contain jsx or will it in the future?

Reply:
oh seeing `<MediaContext.Provider` now, maybe for that one vnode maybe we can use createElement directly?

Reply:
Any reason why? I'm not seeing any clear advantages to your proposal.